ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/HiveRelBuilder.java

```diff
@@ -139,6 +139,10 @@ public static SqlFunction getFloorSqlFunction(TimeUnitRange flag) {
   }
 
   public static SqlAggFunction getRollup(SqlAggFunction aggregation) {
+    IRollupableAggregate x = aggregation.unwrap(IRollupableAggregate.class);
```

Comment:
I added a marker instead of hardcoding it here; we could dissolve the below specializations into the implementations as well